exporters/committime/collector_base.py

```diff
@@ -120,14 +121,53 @@ def generate_metrics(
             watched_namespaces = [
                 namespace.metadata.name for namespace in v1_namespaces.get().items
             ]
+        if watched_namespaces:
+            # Case where comma separated list was provided as set {}
+            watched_namespaces = list(watched_namespaces)
         logging.info("Watching namespaces: %s" % (watched_namespaces))
+        return watched_namespaces
+
+    def _get_openshift_obj_by_app(self, openshift_obj: str, app_label: str) -> dict:
+        apps = []
+
+        # use a jsonpath expression to find all values for the app label
+        jsonpath_str = "$['items'][*]['metadata']['labels']['" + str(app_label) + "']"
+
+        jsonpath_expr = parse(jsonpath_str)
+
+        found = jsonpath_expr.find(openshift_obj)
+
+        apps = [match.value for match in found]
+
+        if not apps:
+            return
```

Comment:
This would return `None`, but the function is annotated as returning a `dict`. We could return an empty dict, or say it returns `Optional[dict]`.

Reply:
Optional[dict] as we want None there, previously it was just continue loop, here we have caller to compare against none. I know comparison will work with empty list, however I prefer to use explicit None.